components/script/script_runtime.rs

```diff
@@ -776,3 +780,44 @@ impl Deref for JSContext {
         &self.0
     }
 }
+
+pub struct StreamConsumer(*mut JSStreamConsumer);
+
+#[allow(unsafe_code)]
+impl StreamConsumer {
+    fn consumeChunk(sc: &mut JSStreamConsumer, begin: *const u8, length: usize) -> bool {
```

Comment:
Let's use a safer [Rust slice](https://doc.rust-lang.org/book/ch04-03-slices.html#other-slices) here instead of separate pointer and length arguments. We'll want `&[u8]`, and we can extract the pointer and length values from that: https://doc.rust-lang.org/stable/std/primitive.slice.html#method.as_ptr

Reply:
Done